src/value.zig

```diff
@@ -1614,6 +1632,16 @@ pub const Value = extern union {
                 };
                 return Value.initPayload(&payload.base);
             },
+            .Union => switch (ty.containerLayout()) {
+                .Auto => unreachable, // Sema is supposed to have emitted a compile error already
+                .Extern => {
+                    @panic("TODO implement readFromMemory for extern union");
```

Comment:
I tried this:
```
                    const union_field = ty.unionFields().values()[0];
                    const tag_val = try Value.Tag.enum_field_index.create(arena, 0);
                    return Tag.@"union".create(arena, .{
                        .tag = tag_val,
                        .val = try readFromMemory(union_field.ty, mod, buffer, arena),
                    });
```
which worked for some smaller things but fails for other stuff.

But again I'm realizing I should probably pick the biggest field here actually instead of the first. Maybe I'll try that later. Would obviously like some feedback on any of this.